### ISSUE TITLE
Create uwrt_mars_rover_hw package to commonize constants and custom controllers

### ISSUE THREAD

**Issue by wmmc88** [uwrobotics/uwrt_mars_rover#179]
yeah. I think its best to keep it as just a string until we create a package for it to avoid duplicate definitions everywhere. Probably like a top level `uwrt_mars_rover_hw` package like what i did for shared custom interfaces in the `ros1` branch

_Originally posted by @wmmc88 in https://github.com/uwrobotics/uwrt_mars_rover/pull/171#discussion_r783400593_

**Comment by wmmc88** [comment]
Not sure if this should be a single package or multiple. Stuff that needs to be added includes iq_current hw state, any sort of shared transmission stuff, shared xacros, custom controllers? 

I could see these being split up into their respective uwrt_mars_rover_description,(xacros), uwrt_mars_rover_hw(shared state handles, shared custom transmissions) and uwrt_mars_rover_controllers packages(shared custom controllers).

**Comment by niiquaye** [comment]
I agree. I really got to learn how you create issues from comments of other PRs - that's just too cool.

**Comment by wmmc88** [comment]
> I agree. I really got to learn how you create issues from comments of other PRs - that's just too cool. 

Right here: https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue#creating-an-issue-from-discussion
